packages/retail-ui-extensions/src/components/DatePicker/DatePicker.ts

```diff
@@ -0,0 +1,15 @@
+import {createRemoteComponent} from '@remote-ui/core';
+
+export interface DatePickerProps {
+  selected?: string;
+  onChange?(selected: string): void;
+  visibleState: [
+    boolean,
+    (setVisibleState: boolean | ((prevState: boolean) => boolean)) => void,
```

Comment:
I'm not sure if this will work with Argo. Have you tried tophatting it, does it work?

Reply:
Seems to work fine, yes

Reply:
The type looks weird because it's a non-React state dispatch `function:React.Dispatch<React.SetStateAction<boolean>>`

which resolves to what you see here

Reply:
Sure!

```
extend('pos.home.modal.render', (root, api) => {
  let datePicker;

  const setVisible = (visible) => {
    datePicker.updateProps({visibleState: [visible, setVisible]})
  }

  datePicker = root.createComponent(
    DatePicker,
    {
      visibleState: [false, setVisible],
      onChange: (date) => {text.appendChild(date)}
    }
  );

  let textContent = '';

  let button = root.createComponent(Button, {title: 'Press to show', onPress: () => setVisible(true)})
  let text = root.createComponent(Text, {}, textContent)

  let screen = root.createComponent(Screen, {name: 'screen', title: 'title'});

  screen.appendChild(button);
  screen.appendChild(text);
  screen.appendChild(datePicker);
  root.appendChild(screen);
  root.mount();
});
```
in javascript
```
const SmartGridModal = () => {
  const [visible, setVisible] = useState(false);
  const [date, setDate] = useState(new Date().toString());
  return (
    <Navigator>
      <Screen name="HelloWorld" title="Hello World!">
        <Button title='Press to show picker' onPress={() => setVisible(true)} />
        <Text>{date}</Text>
        <DatePicker visibleState={[visible, setVisible]} onChange={setDate}/>
      </Screen>
    </Navigator>
  );
  ```
  in react